docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,479 @@
+> addressing inconsistencies in type visibility rules.
+> This change ensures that bounds on type parameters follow the same visibility rules as classes, preventing issues like IR validation errors in JVM.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning for exposing types via type parameter bounds with lower visibility
+> - 2.2.0: raise the warning to an error
+
+### Prohibit inheriting an abstract var property alongside a val property with the same name
+
+> **Issue**: [KT-58659](https://youtrack.jetbrains.com/issue/KT-58659)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: If a class inherits an abstract `var` property from an interface and a `val` property with the same name from a superclass,
+> it now triggers a compilation error. This resolves runtime crashes caused by missing setters in such cases.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning (or an error in progressive mode) when a class inherits an abstract `var` property from an interface and a `val` property with the same name from a superclass
+> - 2.2.0: raise the warning to an error
+
+### Report error when accessing uninitialized enum entries
+
+> **Issue**: [KT-68451](https://youtrack.jetbrains.com/issue/KT-68451)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The compiler now reports an error when uninitialized enum entries are accessed during enum class or entry initialization.
+> This aligns behavior with member property initialization rules, preventing runtime exceptions and ensuring consistent logic.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report an error when accessing uninitialized enum entries
+
+### Changes in K2 smart cast propagation
+
+> **Issue**: [KTLC-34](https://youtrack.jetbrains.com/issue/KTLC-34)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: K2 changes the behavior of smart cast propagation, introducing bidirectional propagation of type
+> information for inferred variables, like `val x = y`. Explicitly typed variables, such as `val x: T = y`,
+> no longer propagate type information, ensuring stricter adherence to declared types.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: enable the new behavior
+
+### Correct the handling of member-extension property overrides in Java subclasses
+
+> **Issue**: [KTLC-35](https://youtrack.jetbrains.com/issue/KTLC-35)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: The getter for member-extension properties overridden by Java subclasses is now hidden in the subclass's scope,
+> aligning its behavior with regular Kotlin properties.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: enable the new behavior
+
+### Correct visibility alignment for getters and setters of var properties overriding a protected val
```

Comment:
Aren't we missing a short summary here?

Reply:
title summarizes it well - so no short summary required